README.md

```diff
@@ -62,6 +62,29 @@ To set up a route, you can do the following:
 ```
 The plugin will render the component server side and return it, where as the route handler will return the props to the frontend when needed.
 
+## Imperatively rendering a component
+Imperative rendering allows routes to dynamically render a component based on specific conditions, instead of always rendering the same component. To do so, we still require that the component is registered to allow Vite to bundle it.
+
+Note that the route which renders the component is a normal route that doesn't need any special route options configuration.
+
+To register a component, you can do the following:
+```js
+  server.registerRenderable(require.resolve('./ImperativelyRenderablePage'))
```

Comment:
`reply.render` taking a string to a path might work in development if you forget to pre-register the renderable -- the API doesn't inform you really that you need to pre-register the components. What do you think about an API where `registerRenderable` returns a token that you then need to pass to `reply.render`, like:

```typescript
const ImperativePage = server.registerRenderable(require.resolve('./ImperativelyRenderablePage'))

server.get('/imperative/:bool', async (request: FastifyRequest<{ Params: { bool: string } }>, reply) => {
    if (request.params.bool == 'true') {
      return reply.render(ImperativePage, {
        hostname: os.hostname(),
        requestIP: request.ip,
      })
    } else {
      return reply.redirect('/not-found')
    }
  })
```